// fawkes-mobile-lib/es/orgSelector/tag/tag.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
require("./tag-sfc.css");
var __vue_render__ = function __vue_render__() {
  var _vm = this;
  var _h = _vm.$createElement;
  var _c = _vm._self._c || _h;
  return _vm.value ? _c('span', {
    staticClass: "tags"
  }, [_c('div', {
    staticClass: "content"
  }, [_vm._v("\n    " + _vm._s(_vm.value) + "\n    "), _vm._v(" "), _vm.closable ? _c('i', {
    staticClass: "fm-icon fm-icon-close",
    nativeOn: {
      "click": function click($event) {
        $event.stopPropagation();
        return _vm.click.apply(null, arguments);
      }
    }
  }) : _vm._e()])]) : _vm._e();
};
var __vue_staticRenderFns__ = [];
var _default2 = {
  render: __vue_render__,
  staticRenderFns: __vue_staticRenderFns__,
  name: "tag",
  props: {
    value: {
      // required:true,
      type: String
    },
    closable: {
      type: Boolean,
      default: function _default() {
        return true;
      }
    }
  },
  methods: {
    click: function click() {
      this.$emit("click", this.value);
    }
  }
};
exports.default = _default2;
