// fawkes-mobile-lib/es/form-cascader/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _babelHelperVueJsxMergeProps = _interopRequireDefault(require("@vue/babel-helper-vue-jsx-merge-props"));
var _utils = require("../utils");
var _field = _interopRequireDefault(require("../field"));
var _popup = _interopRequireDefault(require("../popup"));
var _cascader = _interopRequireDefault(require("../cascader"));
/*
 * @Description: 表单级联选择
 * @Author: ye_xf
 * @Date: 2022-06-10 11:29:59
 * @LastEditTime: 2022-06-14 10:43:09
 * @LastEditors: ye_xf
 * @Reference:
 */

var _createNamespace = (0, _utils.createNamespace)('form-cascader'),
  createComponent = _createNamespace[0];
var _default2 = createComponent({
  props: {
    // Field props
    value: {
      type: String,
      default: ''
    },
    label: {
      type: String,
      default: ''
    },
    name: {
      type: String,
      default: ''
    },
    placeholder: {
      type: String,
      default: '请选择'
    },
    required: {
      type: Boolean,
      default: false
    },
    // Cascader props
    options: {
      type: Array,
      default: function _default() {
        return [];
      }
    },
    fieldNames: {
      type: Object,
      default: function _default() {
        return {
          text: 'text',
          value: 'value',
          children: 'children'
        };
      }
    },
    activeColor: {
      type: String,
      default: '#027aff'
    }
  },
  data: function data() {
    return {
      popupVisible: false
    };
  },
  methods: {
    onFieldClick: function onFieldClick() {
      this.popupVisible = true;
      this.$emit('click');
    },
    onPopupOpen: function onPopupOpen() {
      this.$emit('open');
    },
    onPopupOpened: function onPopupOpened() {
      this.$emit('opened');
    },
    onPopupClose: function onPopupClose() {
      this.$emit('close');
    },
    onPopupClosed: function onPopupClosed() {
      this.$emit('closed');
    },
    onCascaderChange: function onCascaderChange(data) {
      this.$emit('change', data);
    },
    onCascaderFinish: function onCascaderFinish(data) {
      this.popupVisible = false;
      this.$emit('finish', data);
    }
  },
  render: function render() {
    var _attrs,
      _this = this;
    var h = arguments[0];
    return h("div", [h(_field.default, {
      "attrs": (_attrs = {
        "value": this.value,
        "label": this.label,
        "name": this.name,
        "placeholder": this.placeholder,
        "required": this.required,
        "readonly": this.readonly
      }, _attrs["readonly"] = true, _attrs["is-link"] = true, _attrs),
      "on": {
        "click": this.onFieldClick
      }
    }), h(_popup.default, (0, _babelHelperVueJsxMergeProps.default)([{
      "attrs": {
        "visible": this.popupVisible,
        "position": "bottom",
        "round": true
      },
      "on": {
        "open": this.onPopupOpen,
        "opened": this.onPopupOpened,
        "close": this.onPopupClose,
        "closed": this.onPopupClosed
      }
    }, {
      "on": {
        'update:visible': function updateVisible(val) {
          _this.popupVisible = val;
        }
      }
    }]), [h(_cascader.default, {
      "attrs": {
        "title": this.placeholder,
        "options": this.options,
        "active-color": this.activeColor,
        "field-names": this.fieldNames,
        "closeable": false
      },
      "on": {
        "change": this.onCascaderChange,
        "finish": this.onCascaderFinish
      }
    })])]);
  }
});
exports.default = _default2;
