// fawkes-mobile-lib/lib/flow-comments/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _vueCuteTimeline = require("vue-cute-timeline");
var _utils = require("../utils");
/*
 * @Author: gao_m3
 * @Date: 2021-12-21 10:22:55
 * @LastEditors: xie_sm
 * @LastEditTime: 2022-03-31 09:03:23
 * @Descripttion: 
 */

var _createNamespace = (0, _utils.createNamespace)('flow-comments'),
  createComponent = _createNamespace[0],
  bem = _createNamespace[1],
  t = _createNamespace[2];
var _default2 = createComponent({
  components: {
    Timeline: _vueCuteTimeline.Timeline,
    TimelineItem: _vueCuteTimeline.TimelineItem
  },
  props: {
    flowList: {
      type: Array,
      default: function _default() {
        return [];
      }
    }
  },
  mounted: function mounted() {
    console.log(this.flowList);
  },
  methods: {
    // styleChange(obj) {
    //   switch (obj) {
    //     case "提交":
    //       return "progress-submit"
    //     case "同意":
    //       return "progress-agree"
    //     case "退回":
    //       return "progress-back"
    //     case "结束":
    //       return "progress-finish"
    //     default:
    //       return "progress-wait"
    //   }
    // },
    circleColorChange: function circleColorChange(obj) {
      switch (obj) {
        case "提交":
          return "#1790fe";
        case "已办":
          return "#40bb5a";
        case "跳过":
          return "#909399";
        case "委托":
          return "#9f6eff";
        default:
          return "#5ad8e5";
      }
    }
  },
  render: function render() {
    var _this = this;
    var h = arguments[0];
    var flowList = this.flowList;
    return h("timeline", {
      "attrs": {
        "timeline-theme": "#E8EAEB"
      }
    }, [flowList.map(function (item, index) {
      return h("timeline-item", {
        "attrs": {
          "hollow": true,
          "lineColor": _this.circleColorChange(item.approveStateName)
        },
        "key": index
      }, [h("div", {
        "class": "flow-box"
      }, [h("div", {
        "class": "flow-box-title"
      }, [item.taskName]),
      /* <div
       class={"flow-box-state" + ' ' + this.styleChange(item.approveStateName)}
      >
       {item.approveStateName ? item.approveStateName : t('todo')}
      </div> */
      h("div", {
        "class": "flow-box-state"
      }, [h("div", [item.approveDate])]), h("div", {
        "class": "flow-box-line"
      }), h("div", {
        "class": "flow-box-content"
      }, [h("span", [t('operator')]), "\xA0", h("div", {
        "class": "name handle"
      }, [item.asigneeName])]), h("div", {
        "class": "flow-box-content",
        "style": "margin-top: 10px"
      }, [h("span", [t('suggestion')]), "\xA0", h("div", {
        "class": "name handle"
      }, [item.status])]), h("div", {
        "class": "flow-box-taskComment"
      }, [item.comment ? item.comment : t('todo')])])]);
    })]);
  }
});
exports.default = _default2;
