// fawkes-mobile-lib/lib/flow-approve/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _field = _interopRequireDefault(require("../field"));
var _button = _interopRequireDefault(require("../button"));
var _utils = require("../utils");
/*
 * @Author: gao_m3
 * @Date: 2021-12-21 10:22:55
 * @LastEditors: xie_sm
 * @LastEditTime: 2022-03-31 08:50:53
 * @Descripttion: 
 */

var _createNamespace = (0, _utils.createNamespace)('flow-approve'),
  createComponent = _createNamespace[0];
var _default = createComponent({
  props: {
    value: String
  },
  data: function data() {
    return {
      content: ''
    };
  },
  mounted: function mounted() {
    this.content = this.value;
  },
  render: function render() {
    var _this = this;
    var h = arguments[0];
    return h("div", {
      "class": "approve-container"
    }, [h("div", {
      "class": "approve-textarea"
    }, [h("div", {
      "class": "approve-title"
    }, ["\u5BA1\u6279\u610F\u89C1"]), h(_field.default, {
      "attrs": {
        "type": "textarea",
        "maxlength": "100",
        "placeholder": "请输入",
        "show-word-limit": true
      },
      "model": {
        value: _this.content,
        callback: function callback($$v) {
          _this.content = $$v;
        }
      }
    })]), h("div", {
      "class": "btn"
    }, [h(_button.default, {
      "class": "submit-btn",
      "attrs": {
        "type": "primary"
      },
      "on": {
        "click": function click() {
          _this.$emit('onSubmit', _this.content);
        }
      }
    }, [this.t('submit')]), h(_button.default, {
      "class": "reject-btn",
      "attrs": {
        "plain": true,
        "type": "default"
      },
      "on": {
        "click": function click() {
          _this.$emit('onReject');
        }
      }
    }, [this.t('reject')])])]);
  }
});
exports.default = _default;
