// fawkes-mobile-lib/lib/steps/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _utils = require("../utils");
var _relation = require("../mixins/relation");
/*
 * @Author: gao_m3
 * @Date: 2021-11-30 16:09:57
 * @LastEditors:
 * @LastEditTime: 2021-12-14 16:19:54
 * @Descripttion:
 */

var _createNamespace = (0, _utils.createNamespace)('steps'),
  createComponent = _createNamespace[0],
  bem = _createNamespace[1];
var _default = createComponent({
  mixins: [(0, _relation.ParentMixin)('fmSteps')],
  props: {
    iconPrefix: String,
    finishIcon: String,
    activeColor: String,
    inactiveIcon: String,
    inactiveColor: String,
    active: {
      type: [Number, String],
      default: 0
    },
    direction: {
      type: String,
      default: 'horizontal'
    },
    activeIcon: {
      type: String,
      default: 'checked'
    }
  },
  render: function render() {
    var h = arguments[0];
    return h("div", {
      "class": bem([this.direction])
    }, [h("div", {
      "class": bem('items')
    }, [this.slots()])]);
  }
});
exports.default = _default;
