// fawkes-mobile-lib/lib/mixins/portal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PortalMixin = PortalMixin;
function getElement(selector) {
  if (typeof selector === 'string') {
    return document.querySelector(selector);
  }
  return selector();
}
function PortalMixin(_temp) {
  var _ref = _temp === void 0 ? {} : _temp,
    ref = _ref.ref,
    afterPortal = _ref.afterPortal;
  return {
    props: {
      appendToBody: {
        default: true,
        type: Boolean
      },
      getContainer: [String, Function]
    },
    watch: {
      getContainer: 'portal'
    },
    mounted: function mounted() {
      if (this.getContainer) {
        this.portal();
      }
    },
    methods: {
      portal: function portal() {
        var appendToBody = this.appendToBody,
          getContainer = this.getContainer;
        var el = ref ? this.$refs[ref] : this.$el;
        var container;
        var gc = appendToBody ? 'body' : getContainer;
        if (gc) {
          container = getElement(gc);
        } else if (this.$parent) {
          container = this.$parent.$el;
        }
        if (container && container !== el.parentNode) {
          container.appendChild(el);
        }
        if (afterPortal) {
          afterPortal.call(this);
        }
      }
    }
  };
}
