// fawkes-lib/lib/mixins/selector.js
'use strict';

exports.__esModule = true;
/*
 * @Author: wei_jt@ecidi.com
 * @Date: 2019-07-19 19:40:56
 * @LastEditors: Please set LastEditors
 * @LastEditTime: 2021-03-09 16:12:13
 * @Description: 选择组件通用部分
 */
exports.default = {
  inject: {
    fksForm: {
      default: function _default() {
        return {};
      }
    },
    fksFormItem: {
      default: function _default() {
        return {};
      }
    }
  },
  props: {
    showAllUser: {
      type: Boolean,
      default: false
    },
    width: {
      type: String,
      default: '100%'
    },
    url: {
      type: String,
      default: ''
    },
    params: {
      type: Object,
      default: function _default() {
        return {};
      }
    },
    headers: {
      type: Object,
      default: function _default() {
        return {};
      }
    },
    multiple: Boolean,
    multipleLimit: {
      type: Number,
      default: 0
    },
    readonly: Boolean,
    disabled: Boolean,
    /** 标签是否收缩 */
    collapseTags: Boolean,
    size: String,
    related: {
      type: Object,
      default: function _default() {
        return {};
      }
    }
  },
  data: function data() {
    return {
      initialInputHeight: 0,
      visible: false
    };
  },

  computed: {
    collapseTagSize: function collapseTagSize() {
      return ['small', 'mini'].indexOf(this.size) > -1 ? 'mini' : 'small';
    },
    forbidden: function forbidden() {
      return this.disabled || !!(this.fksForm && this.fksForm.disabled) || ((this.fksForm || {}).flowConfig || {})[(this.fksFormItem || {}).prop] == 'disabled';
    },
    uploadReadOnly: function uploadReadOnly() {
      return this.readonly || ((this.fksForm || {}).flowConfig || {})[(this.fksFormItem || {}).prop] == 'readonly';
    }
  },
  methods: {
    resetInputHeight: function resetInputHeight() {
      var _this = this;

      this.$nextTick(function () {
        if (!_this.$refs.input) return;
        var inputChildNodes = _this.$refs.input.$el.childNodes;
        var input = [].filter.call(inputChildNodes, function (item) {
          return item.tagName === 'INPUT';
        })[0];
        var tags = _this.$refs.tags;
        var sizeInMap = _this.initialInputHeight || 36;
        var list = _this.sureList || _this.chosedList;
        input.style.height = list.length === 0 ? sizeInMap + 'px' : Math.max(tags ? tags.clientHeight + (tags.clientHeight > sizeInMap ? 4 : 0) : 0, sizeInMap) + 'px';
      });
    }
  },
  watch: {
    collapseTags: function collapseTags() {
      if (this.multiple) {
        this.resetInputHeight();
      }
    }
  },
  mounted: function mounted() {
    var sizeMap = {
      large: 40,
      small: 32,
      mini: 28
    };
    this.initialInputHeight = sizeMap[this.size];
  }
};
