// fawkes-lib/lib/utils/types.js
'use strict';

exports.__esModule = true;
exports.isString = isString;
exports.isObject = isObject;
exports.isHtmlElement = isHtmlElement;
/*
 * @Author: gao_m3
 * @Date: 2021-01-06 09:18:03
 * @LastEditors: 
 * @LastEditTime: 2021-04-02 09:09:03
 * @Descripttion: 
 */
function isString(obj) {
  return Object.prototype.toString.call(obj) === '[object String]';
}

function isObject(obj) {
  return Object.prototype.toString.call(obj) === '[object Object]';
}

function isHtmlElement(node) {
  return node && node.nodeType === Node.ELEMENT_NODE;
}

var isFunction = exports.isFunction = function isFunction(functionToCheck) {
  var getType = {};
  return functionToCheck && getType.toString.call(functionToCheck) === '[object Function]';
};

var isUndefined = exports.isUndefined = function isUndefined(val) {
  return val === void 0;
};

var isDefined = exports.isDefined = function isDefined(val) {
  return val !== undefined && val !== null;
};
var isNumber = exports.isNumber = function isNumber(val) {
  return typeof val === 'number';
};
var isArray = exports.isArray = function isArray(val) {
  return Array.isArray(val);
};
