// fawkes-mobile-lib/es/collapse/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _utils = require("../utils");
var _relation = require("../mixins/relation");
var _constant = require("../utils/constant");
/*
 * @Author: gao_m3
 * @Date: 2021-11-30 16:09:54
 * @LastEditors:
 * @LastEditTime: 2021-12-16 10:48:16
 * @Descripttion:
 */

var _createNamespace = (0, _utils.createNamespace)('collapse'),
  createComponent = _createNamespace[0],
  bem = _createNamespace[1];
var _default = createComponent({
  mixins: [(0, _relation.ParentMixin)('fmCollapse')],
  props: {
    accordion: Boolean,
    value: [String, Number, Array],
    border: {
      type: Boolean,
      default: true
    }
  },
  methods: {
    switch: function _switch(name, expanded) {
      if (!this.accordion) {
        name = expanded ? this.value.concat(name) : this.value.filter(function (activeName) {
          return activeName !== name;
        });
      }
      this.$emit('change', name);
      this.$emit('input', name);
    }
  },
  render: function render() {
    var _ref;
    var h = arguments[0];
    return h("div", {
      "class": [bem(), (_ref = {}, _ref[_constant.BORDER_TOP_BOTTOM] = this.border, _ref)]
    }, [this.slots()]);
  }
});
exports.default = _default;
